### PR TITLE
add spring-cloud-gateway charts

### PR DESCRIPTION
**Pull request by Accelerator96**

### CODE REVIEW

File: charts/opentelemetry-demo-lite/templates/spring-cloud-gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                containerPort: 18080
+                protocol: TCP
+            env:
+              - name: ADSERVICE_ADDR
+                value: 'http://{{ include "otel-demo.name" . }}-adservice:8081'
+              - name: AUTH_ENABLED
+                value: "false"
+            resources:
+              limits:
+                memory: 512Mi
```
Comment (panpan0000) [inline charts/opentelemetry-demo-lite/templates/spring-cloud-gateway.yaml:37]: is 512MB enough?

**Comment by panpan0000** [comment]
good job

**Comment by panpan0000** [comment]
漏了东西?

```
Error: UPGRADE FAILED: values don't meet the specifications of the schema(s) in the following chart(s):
opentelemetry-demo:
- components: Additional property springCloudGateway is not allowed

helm.go:84: [debug] values don't meet the specifications of the schema(s) in the following chart(s):
opentelemetry-demo:
- components: Additional property springCloudGateway is not allowed

```

File: charts/opentelemetry-demo-lite/values.yaml [hunk 2]
```diff
@@ -414,6 +414,13 @@ opentelemetry-demo:
           cpu: "0.5"
           memory: 512Mi
 
+    springCloudGateway:
+      enabled: false
+      image:
+        repository: ghcr.io/openinsight-proj/demo
```
Comment (panpan0000) [inline charts/opentelemetry-demo-lite/values.yaml:420]: 镜像`ghcr.io/openinsight-proj/demo:6adde5369780c1be22f148e0790a080575827aee-springCloudGateway` 拉不到的

**Review by panpan0000 — CHANGES_REQUESTED** [review]
for above comments

File: charts/opentelemetry-demo-lite/templates/spring-cloud-gateway.yaml (hunk 1 above)
Comment (Accelerator96) [reply charts/opentelemetry-demo-lite/templates/spring-cloud-gateway.yaml:37]: 512m enough

File: charts/opentelemetry-demo-lite/templates/springCloudGateway/spring-cloud-gateway.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Release.Name }}-spring-cloud-gateway
+spec:
+  type: {{ .Values.extensions.springCloudGateway.service.type | default "ClusterIP" }}
+  ports:
+    - name: http
+      port: 18081
```
Comment (panpan0000) [inline charts/opentelemetry-demo-lite/templates/springCloudGateway/spring-cloud-gateway.yaml:47]: !! 上面应用暴露18080， 这里的svc却是18081

**Comment by JaredTan95** [comment]
@Accelerator96 pls update spring cloud gateway image with latest tag: https://github.com/openinsight-proj/opentelemetry-demo/actions/runs/4815719682/jobs/8574661938

**Review by JaredTan95 — APPROVED** [review]

**Pull request merged**